### PR TITLE
New modules: Fgbio callmolecularconsensusreads and sortbam

### LINKED ISSUE #188: new module: fgbio/callmolecularconsensusreads
I think it would be good to have module for [fgbio](https://bioconda.github.io/recipes/fgbio/README.html)

### PR DESCRIPTION
**Pull request by sruthipsuresh**
## PR checklist

Closes #188 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `<SOFTWARE>.version.txt` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

Added callmolecularconsensusreads and sortbam modules (since the molecularconsensusreads requires the input file being processed in the sortbam module). Please let me know if any changes are to be made, or if the sortbam step can be used in the script of the callmolecularconsensusreads process instead.

### CODE REVIEW

File: tests/software/fgbio/callmolecularconsensusreads/main.nf [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [args: '-s TemplateCoordinate', suffix: '_out'] )
+include { FGBIO_CALLMOLECULARCONSENSUSREADS } from '../../../../software/fgbio/callmolecularconsensusreads/main.nf' addParams( options: [args: '-M 1', suffix: '_molreads'] )
+
+workflow test_fgbio_callmolecularconsensusreads {
+    
+    def input = []
+    input = [ [ id:'test' ], // meta map
+              file('https://github.com/fulcrumgenomics/fgbio/blob/master/src/test/resources/com/fulcrumgenomics/bam/200reads.bam?raw=true', checkIfExists: true) ]
```
Comment (grst) [inline tests/software/fgbio/callmolecularconsensusreads/main.nf:12]: Please use the test files provided in the modules respository. Please have a look at `tests/config/test_data.config` and use `params.test_data` to access the files.

File: tests/software/fgbio/sortbam/main.nf [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [:] )
+
+workflow test_fgbio_sortbam {
+    
+    def input = []
+    input = [ [ id:'test' ], // meta map
+              file('https://github.com/fulcrumgenomics/fgbio/blob/master/src/test/resources/com/fulcrumgenomics/bam/200reads.bam?raw=true', checkIfExists: true) ]
```
Comment (grst) [inline tests/software/fgbio/sortbam/main.nf:11]: same as above

**Review by grst — CHANGES_REQUESTED** [review]
Just a small issue with the test data, otherwise LGTM :rocket:

File: tests/software/fgbio/callmolecularconsensusreads/main.nf (hunk 1 above)
Comment (sruthipsuresh) [reply tests/software/fgbio/callmolecularconsensusreads/main.nf]: Thank you for the feedback! I fixed the data for both modules.

**Comment by sruthipsuresh** [comment]
Thanks for the feedback! I went ahead and changed the test data. Please let me know if there are any other revisions to be made. @grst

File: software/fgbio/callmolecularconsensusreads/meta.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        documentation: http://fulcrumgenomics.github.io/fgbio/
+        licence: ['MIT']
+
+input:
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false, collapse:false ]
+    - reads:
```
Comment (drpatelh) [inline software/fgbio/callmolecularconsensusreads/meta.yml:22]:
```suggestion
    - bam:
```

File: software/fgbio/callmolecularconsensusreads/meta.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          The input SAM or BAM file.
+        pattern: "*.{bam,sam}"
+
+output:
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false ]
+    - reads:
```
Comment (drpatelh) [inline software/fgbio/callmolecularconsensusreads/meta.yml:34]:
```suggestion
    - bam:
```

File: software/fgbio/sortbam/main.nf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    fgbio \\
+        SortBam \\
+        -i $bam \\
+        $options.args \\
+        -o ${prefix}.bam
+    fgbio --version | sed -e "s/fgbio v//g" > ${software}.version.txt
+    """
+
```
Comment (drpatelh) [inline software/fgbio/sortbam/main.nf:38]:
```suggestion
```

File: software/fgbio/sortbam/meta.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        documentation: http://fulcrumgenomics.github.io/fgbio/
+        licence: ['MIT']
+
+input:
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false, collapse:false ]
+    - reads:
```
Comment (drpatelh) [inline software/fgbio/sortbam/meta.yml:20]:
```suggestion
    - bam:
```

File: software/fgbio/sortbam/meta.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          The input SAM or BAM file to be sorted.
+        pattern: "*.{bam,sam}"
+
+output:
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false ]
+    - reads:
```
Comment (drpatelh) [inline software/fgbio/sortbam/meta.yml:32]:
```suggestion
    - bam:
```

File: tests/software/fgbio/callmolecularconsensusreads/main.nf [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [args: '-s TemplateCoordinate', suffix: '_out'] )
+include { FGBIO_CALLMOLECULARCONSENSUSREADS } from '../../../../software/fgbio/callmolecularconsensusreads/main.nf' addParams( options: [args: '-M 1', suffix: '_molreads'] )
+
+workflow test_fgbio_callmolecularconsensusreads {
+    
+    def input = []
```
Comment (drpatelh) [inline tests/software/fgbio/callmolecularconsensusreads/main.nf:10]:
```suggestion
```

File: tests/software/fgbio/callmolecularconsensusreads/main.nf [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [args: '-s TemplateCoordinate', suffix: '_out'] )
+include { FGBIO_CALLMOLECULARCONSENSUSREADS } from '../../../../software/fgbio/callmolecularconsensusreads/main.nf' addParams( options: [args: '-M 1', suffix: '_molreads'] )
+
+workflow test_fgbio_callmolecularconsensusreads {
+
```
Comment (drpatelh) [inline tests/software/fgbio/callmolecularconsensusreads/main.nf:9]:
```suggestion
```

File: tests/software/fgbio/callmolecularconsensusreads/main.nf [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [args: '-s TemplateCoordinate', suffix: '_out'] )
+include { FGBIO_CALLMOLECULARCONSENSUSREADS } from '../../../../software/fgbio/callmolecularconsensusreads/main.nf' addParams( options: [args: '-M 1', suffix: '_molreads'] )
+
+workflow test_fgbio_callmolecularconsensusreads {
+    
+    def input = []
+    input = [ [ id:'test' ], // meta map
+                file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true), ]
```
Comment (drpatelh) [inline tests/software/fgbio/callmolecularconsensusreads/main.nf:12]:
```suggestion
                file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true) ]
```

File: tests/software/fgbio/sortbam/main.nf [hunk 11]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [:] )
+
+workflow test_fgbio_sortbam {
+
```
Comment (drpatelh) [inline tests/software/fgbio/sortbam/main.nf:8]:
```suggestion
```

File: tests/software/fgbio/sortbam/main.nf [hunk 12]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { FGBIO_SORTBAM } from '../../../../software/fgbio/sortbam/main.nf' addParams( options: [:] )
+
+workflow test_fgbio_sortbam {
+    
+    def input = []
```
Comment (drpatelh) [inline tests/software/fgbio/sortbam/main.nf:9]:
```suggestion
```

File: tests/software/fgbio/sortbam/main.nf [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    input = [ [ id:'test' ], // meta map
+              file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true), ]
```
Comment (drpatelh) [inline tests/software/fgbio/sortbam/main.nf:11]:
```suggestion
              file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ]
```

File: tests/software/fgbio/sortbam/test.yml [hunk 14]
```diff
@@ -0,0 +1,10 @@
+- name: fgbio sortbam
+  command: nextflow run tests/software/fgbio/sortbam -entry test_fgbio_sortbam -c tests/config/nextflow.config
+  tags:
+    - fgbio_sortbam
+    - fgbio
+  files:
+    - path: output/fgbio/test.bam
+      md5sum: 56987f74214cdc72ae840e0007ff4935
+
```
Comment (drpatelh) [inline tests/software/fgbio/sortbam/test.yml:9]:
```suggestion
```

File: tests/software/fgbio/sortbam/test.yml [hunk 15]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (drpatelh) [inline tests/software/fgbio/sortbam/test.yml:10]:
```suggestion

```

**Review by drpatelh — COMMENTED** [review]
Looks awesome @sruthipsuresh 😍 Couple of very very minor changes.

**Review by drpatelh — APPROVED** [review]

**Pull request merged**